src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -10171,7 +10171,8 @@ void Unit::StopMoving()
         return;
 
     // Update position now since Stop does not start a new movement that can be updated later
-    UpdateSplinePosition();
+    if (movespline->Started())
```

Comment:
isn't !movespline->Finalized()  the common helper used for checking if a spline is currently in progress?

Reply:
Does that tell if the spline started yet ?

Reply:
It does return true as long as a spline is currently in effect, yes. It's the common check used in every movement generator to check if a creature is actually moving or not. (It's also the common check to protect against nullptrs: no spline in effect --> movespline will return nullptr)

Reply:
But I don't need to know if it's in effect or finished,  I need to know if the Unit walked at least 1 step already in three current spline (hence the name Started and not something like InProgress)